### DISCUSSION TITLE
Spotting the New Publisher/Subscriber - RTPS

### DISCUSSION THREAD

**Discussion by MuhammedErkam** [OpenDDS/OpenDDS#3842] [Q&A] [answered]
How does OpenDDS distinguish when new publisher or subscriber is added in RTPS Configuration?

**Comment by mitza-oci** [comment, marked as answer]
Two features from the DDS standard may help with this: Built-in Topics and the "Publication/Subscription Matched" Communication Statuses.  The application can access DataReaders for the Built-in Topics and the Communication Statuses can be used to trigger listeners and /or conditions.

**Reply by MuhammedErkam** [reply]
Are SPDP and SEDP relevant to this property? Or only are PublicationMatchedStatus and SubscriptionMatchedStatus used for this property?

**Reply by mitza-oci** [reply]
Both of the features I mentioned before, namely Built-in Topics and the "Publication/Subscription Matched" Communication Statuses are part of the public and spec-defined API.  SPDP and SEDP are underlying protocols used by RTPS Discovery.
